cmd/keeper/cmd/keeper.go

```diff
@@ -809,20 +811,56 @@ func (p *PostgresKeeper) resync(db, followedDB *cluster.DB, tryPgrewind bool) er
 	// postgresql version is > 9.5 since someone can also use an externally
 	// installed pg_rewind for postgres 9.4. If a pg_rewind executable
 	// doesn't exists pgm.SyncFromFollowedPGRewind will return an error and
-	// fallback to pg_basebackup
+	// fallback to pg_basebackup. If there is no pg_rewind timeout or
+	// interval set, we will only try pg_rewind once and fallback to
+	// pg_basebackup straight away. Otherwise we will keep retrying until
+	// pg_rewind succeeds or we reach the timeout (and then we will
+	// fallback to pg_basebackup).
 	if tryPgrewind && p.usePgrewind(db) {
-		connParams := p.getSUConnParams(db, followedDB)
-		log.Infow("syncing using pg_rewind", "followedDB", followedDB.UID, "keeper", followedDB.Spec.KeeperUID)
-		// TODO: Make the forceCheckpoint parameter use cluster specification
-		if err := pgm.SyncFromFollowedPGRewind(connParams, p.pgSUPassword, true); err != nil {
-			// log pg_rewind error and fallback to pg_basebackup
-			log.Errorw("error syncing with pg_rewind", zap.Error(err))
+		var err error
+		timeout := db.Spec.PgrewindTimeout.Duration
+		interval := db.Spec.PgrewindInterval.Duration
+		checkpoint := db.Spec.PgrewindCheckpoint
+
+		pgrewind := func() error {
+			connParams := p.getSUConnParams(db, followedDB)
+			log.Infow("syncing using pg_rewind", "followedDB", followedDB.UID, "keeper", followedDB.Spec.KeeperUID)
+			if err = pgm.SyncFromFollowedPGRewind(connParams, p.pgSUPassword, checkpoint); err != nil {
+				log.Errorw("error syncing with pg_rewind", zap.Error(err))
+				pgm.SetRecoveryParameters(p.createRecoveryParameters(true, standbySettings, nil, nil))
+			}
+			return err
+		}
+
+		// Guarantee we run pg_rewind at least once if usePgrewind is set to
+		// true - this maintains the behaviour from before retrying of pg_rewind
+		// was supported
+		if timeout == 0 || interval == 0 {
+			if err = pgrewind(); err == nil {
+				return nil
+			}
 		} else {
-			pgm.SetRecoveryParameters(p.createRecoveryParameters(true, standbySettings, nil, nil))
-			return nil
+			// Retry pg_rewind at an interval with jitter until it succeeds or
+			// we timeout retrying
+			timeoutAfter := time.After(timeout)
+			jitter := time.Duration(rand.Int63n(int64(interval)) / 4)
+			intervalTick := time.Tick(interval + jitter)
+
+		Rewind:
+			for true {
```

Comment:
Using `for true` instead of just `for` seems to be the convention in the codebase.